.github/workflows/push.yml

```diff
@@ -0,0 +1,23 @@
+name: Push
+on:
+  branch_protection_rule:
+  push:
+    branches: [ main ]
+    paths-ignore:
+      - 'doc/**'
+      - 'enhancements/**'
+      - '*.md'
+
+jobs:
+  # for each PR merge, trigger image build
```

Comment:
Since we have pull_request.yml to check candidate merge PRs, and we also have daily.yaml to build image and check daily, do we still need push.yml here?

Reply:
Daily build and PR level build are two different things, why not?

Reply:
I mean push_request.yml and push.yml.

Reply:
pr, push, daily are different things.
For each PR, 
- pr means before the PR been merged. For example, we run lint before each PR and after the PR been merged the code passed the lint hence we don't need to run another turn for lint after the merge at push level or daily level.
- push means after the PR been merged. At push level, as we refresh test coverage, and build the image base on latest code.
- daily(or any schedule) is a daily build on main/default/specific branch. Take build image as an example, as we don't have PR merge per day, hence we have the image daily build to update dependences to latest as daily build. Same for go sec and score card.... SBOM as it relays on some place to store the result, so currently we make it pre releases.(store with release page on github) further if we find a place, we can make it into daily build as well.